cli/cppcheckexecutor.cpp

```diff
@@ -158,7 +158,33 @@ bool CppCheckExecutor::parseFromArgs(CppCheck *cppcheck, int argc, const char* c
 #else
     const bool caseSensitive = true;
 #endif
-    if (!pathnames.empty()) {
+    if (!mSettings->project.fileSettings.empty()) {
+        // filter only for the selected filenames from all project files
+        std::list<ImportProject::FileSettings> newList;
+
+        for (const std::string fname : pathnames) {
+            for (const ImportProject::FileSettings fsetting : settings.project.fileSettings)
```

Comment:
I don't believe we need to copy fileSettings items.. so please make fsetting a reference.